src/test/features/offer/form/models/offer.ts

```diff
@@ -0,0 +1,88 @@
+import { expect } from 'chai'
+import { Validator } from 'class-validator'
+import { expectValidationError } from '../../../../app/forms/models/validationUtils'
+import { Offer } from 'offer/form/models/offer'
+import { ValidationErrors } from 'offer/form/models/offer'
+import { LocalDate } from 'forms/models/localDate'
+import * as moment from 'moment'
+
+describe('Offer', () => {
+  describe('constructor', () => {
+    it('should set the fields to undefined', () => {
+      const offer = new Offer()
+      expect(offer.offerText).to.be.equal(undefined)
+      expect(offer.completionDate).to.be.equal(undefined)
+    })
+  })
+
+  describe('form object deserialization', () => {
+    it('should return undefined when value is undefined', () => {
+      expect(Offer.fromObject(undefined)).to.be.equal(undefined)
+    })
+
+    it('should return null when value is null', () => {
+      expect(Offer.fromObject(null)).to.be.equal(null)
+    })
+
+    it('should leave missing fields undefined', () => {
+      expect(Offer.fromObject({})).to.deep.equal(new Offer())
+    })
+
+    it('should deserialize all fields', () => {
+      const date = new LocalDate()
+      expect(Offer.fromObject({ offerText: 'offer Text', completionDate: date })).to.deep.equal(new Offer('offer Text', date))
+    })
+  })
+
+  describe('deserialization', () => {
+    it('should return instance initialised with defaults given undefined', () => {
+      expect(new Offer().deserialize(undefined)).to.deep.equal(new Offer())
+    })
+
+    it('should return instance initialised with defaults when given null', () => {
+      expect(new Offer().deserialize(null)).to.deep.equal(new Offer())
+    })
+
+    it('should return instance with set fields from given object', () => {
+      const date = new LocalDate()
+      expect(new Offer().deserialize({ offerText: 'offer Text', completionDate: date.asString() })).to.deep.equal(new Offer('offer Text', date))
+    })
+  })
+
+  describe('validation', () => {
+    const validator: Validator = new Validator()
+
+    describe('should reject when', () => {
+      it('undefined offer text', () => {
+        const futureDate = moment().add(10, 'days')
+        const date = new LocalDate(futureDate.year(), futureDate.month(), futureDate.day())
+        const errors = validator.validateSync(new Offer(undefined, date))
+        expect(errors.length).to.equal(2)
+        expectValidationError(errors, ValidationErrors.OFFER_REQUIRED)
+        expectValidationError(errors, ValidationErrors.OFFER_TEXT_TOO_LONG.replace('$constraint1','99000'))
```

Comment:
How's it possible that it displays both "required" and "too long" validation errors for blank value? Shouldn't it be just "required"?